internal/kubernetes/operator/project/project.go

```diff
@@ -835,6 +800,75 @@ func buildAlertConfigurations(acProvider store.AlertConfigurationLister, project
 	return results, secretResults, nil
 }
 
+func convertMatchers(atlasMatcher []map[string]interface{}) []akov2.Matcher {
+	res := make([]akov2.Matcher, 0, len(atlasMatcher))
+	for _, m := range atlasMatcher {
+		if matcher, err := toMatcher(m); err == nil {
+			res = append(res, matcher)
+		}
+	}
+	return res
+}
+
+func toMatcher(m map[string]interface{}) (akov2.Matcher, error) {
+	var matcher akov2.Matcher
+	if m == nil {
+		return matcher, fmt.Errorf("nil map cannot be converted to Matcher")
+	}
+	fieldName, err := keyAsString(m, "FieldName")
+	if err != nil {
+		return matcher, err
+	}
+	operator, err := keyAsString(m, "Operator")
+	if err != nil {
+		return matcher, err
+	}
+	value, err := keyAsString(m, "Value")
+	if err != nil {
+		return matcher, err
+	}
+	matcher.FieldName = fieldName
+	matcher.Operator = operator
+	matcher.Value = value
+	return matcher, nil
+}
+
+func keyAsString(m map[string]interface{}, key string) (string, error) {
+	v, ok := m[key]
+	if !ok {
+		return "", fmt.Errorf("no key %q at map %v", key, m)
+	}
+	s, ok := (v).(string)
+	if !ok {
+		return "", fmt.Errorf("%q is not a string at map %v", key, m)
```

Comment:
this will produce an error if there is no string value set for the given key, even if it is `nil`, aka unset. I am not convinced we should error out here but rather gracefully continue with other matchers.

Reply:
clarified in other threads.